eureka2-client/src/main/java/com/netflix/eureka2/client/functions/ChangeNotificationFunctions.java

```diff
@@ -0,0 +1,88 @@
+package com.netflix.eureka2.client.functions;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.ChangeNotification.Kind;
+import com.netflix.eureka2.utils.rx.RxFunctions;
+import rx.Observable;
+import rx.Observable.Transformer;
+import rx.functions.Func1;
+
+/**
+ * @author Tomasz Bak
+ */
+public final class ChangeNotificationFunctions {
+
+    private ChangeNotificationFunctions() {
+    }
+
+    /**
+     * Convert change notification stream with buffering sentinels into stream of lists, where each
+     * list element contains a batch of data delineated by the markers.
+     *
+     * @return observable of non-empty list objects
+     */
+    public static <T> Transformer<ChangeNotification<T>, List<ChangeNotification<T>>> buffers() {
```

Comment:
What is the contract for this operations if we received two successive BufferSentinels from the stream? Should we emit empty lists or emit nothing (what the code seem to do right now)? We should document the expected behaviour in this case.